src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3677,36 +3676,46 @@ size_t os::Linux::find_large_page_size(size_t large_page_size) {
     if (entry->d_type == DT_DIR &&
         sscanf(entry->d_name, "hugepages-%zukB", &page_size) == 1) {
       // The kernel is using kB, hotspot uses bytes
-      if (large_page_size == page_size * K) {
-        closedir(dir);
-        return large_page_size;
+      if (page_size * K > (size_t)Linux::page_size()) {
```

Comment:
I do not think excluding the base page size is needed here. The directory only contains entries for huge pages. If for any weird reason this is the same as the base page size (which I have never seen) I would include it, since its a huge page too. But I do not think this can happen.